apps/api/lib/helpers/verifyCredentialSyncEnabled.ts

```diff
@@ -0,0 +1,24 @@
+import type { NextMiddleware } from "next-api-middleware";
+
+import { APP_CREDENTIAL_SHARING_ENABLED } from "@calcom/lib/constants";
+
+export const verifyCredentialSyncEnabled: NextMiddleware = async (req, res, next) => {
+  const { isAdmin } = req;
+
+  if (!isAdmin) {
+    return res.status(401).json({ error: "Only admin API keys can access credential syncing endpoints" });
```

Comment:
Should this actually be a 403? If it reaches here does that mean we've authenticated the request based on the key supplied but they just don't have access? If so, should be a 403.